50-dhole/technical_specification.md

```diff
@@ -240,10 +240,141 @@ B. For merging relation properties:
   ```
 
 ### Refactor aggregate transformation:
-TBD.
+
+The former `aggregate` transformation shall be replaced by a mixture of existing
+and to be written transformations. The following new transformations shall be
+implemented:
+
+#### Transformation 1: Add Content Property
+
+A transformation that enables the insertion of a subschema into a new property
+of any object within the current content schema of a class, including an initial
+value. The schema will be modified by inserting the subschema at the specified path. The data will be modified by inserting the default value at the specified path. The user may control whether or not the newly added property is added to the parent property's `required` list.
+
+##### Configuration
+
+```yaml
+# Generic Example, defaults shown for optional parameters
+- class_name: MyClass
+  target_content:
+    object_path: "" # Optional, default empty string refers to content root object
+    property_name: new_property
+    required: true # Optional
+  schema: # Optional
+    type: "object"
+    additionalProperties: false
+  value: {} # Optional
+
+# Examples from current config
+- class_name: Dataset
+  target_content:
+    property_name: datasets
+
+- class_name: Dataset
+  target_content:
+    property_name: "studies_summary"
+```
+
+#### Transformation 2: Count References
+
+* The transformation shall count how many target objects are referenced from each source object given the reference name.
+* The transformation shall validate whether the target is defined with multiplicity and fail otherwise
+
+Example config:
+
+```yaml
+- class_name: Dataset
+  target_content:
+    object_path: samples_summary
+    property_name: count
+  source_relation_path: "Dataset(samples)>Sample"
```

Comment:
Great, actually I just thought that in deed passive relations (containing <) might also be interesting to count. This notation would enable it.